analysis_driver/quality_control/median_coverage.py

```diff
@@ -0,0 +1,74 @@
+import os
+from analysis_driver import executor
+from analysis_driver.config import default as cfg
+from analysis_driver.notification import default as ntf
+from .quality_control_base import QualityControl
+
+
+class SamtoolsDepth(QualityControl):
+    def __init__(self, dataset, working_dir, bam_file):
+        super().__init__(dataset, working_dir)
+        self.bam_file = bam_file
+        self.working_dir =  working_dir
+        self.exception = None
+
+
+    def _get_samtools_depth_command(self):
+        samtools_bin = cfg['tools']['samtools']
+        samtools_depth_out_file = ((self.bam_file).rstrip('bam') + 'depth')
+        samtools_depth_command = "%s depth -a -a -q 0 -Q 0 %s > %s" % (samtools_bin, self.bam_file, samtools_depth_out_file)
+        return samtools_depth_command, samtools_depth_out_file
+
+
+
+    def _run_samtools_depth(self):
+        """
+        :return string: the expected outfile from samtools depth
+        """
+        samtools_depth_command, samtools_depth_out_file = self._get_samtools_depth_command()
+        ntf.start_stage('run_samtools_depth')
+        samtools_depth_executor = executor.execute(
+            [samtools_depth_command],
+            job_name='samtoolsdepth',
+            working_dir=self.working_dir,
+            cpus=2,
+            mem=10
+        )
+        exit_status = samtools_depth_executor.join()
+        ntf.end_stage('run_samtools_depth', exit_status)
+        return samtools_depth_out_file
+
+    def _get_depth_histogram_command(self):
+        samtools_depth_outfile = self._run_samtools_depth()
+        depth_histogram_outfile = samtools_depth_outfile.rstrip('depth') + 'hist'
+        depth_histogram_command = "awk -F '\t' '{print $3}' %s | sort | uniq -c | sort -nr | sort -k2 -n > %s" % (samtools_depth_outfile,
+                                                                                                    depth_histogram_outfile)
```

Comment:
What about:
```bash
awk -F '\t' '{array[$1"\t"$3]+=1} END{for (val in array){print val"\t"array[val]}}' | sort -k 1,1 -nk 2,2
```
The format would be a bit different but it would be faster and you would have all the chromosomes  
Alternatively 
```bash
awk -F '\t' '{array[$3]+=1} END{for (val in array){print array[val]"\t"val}}' | sort -nk 2,2
```
would give the same results but faster